src/main/webapp/js/components/app.jsx

```diff
@@ -1,15 +1,52 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { HashRouter as Router, Route, Switch } from 'react-router-dom';
 
 import 'bootstrap/dist/css/bootstrap.min.css';
 import '../../css/main.scss';
 
-import Navbar from './navbar';
+import AppContent from './app-content';
 import loadIcons from '../utils/icon-loader';
+import { fetchAllFieldConfigs, fetchEntity } from '../actions/index';
+import { PERMISSION_ENTITY, LANGUAGE_ENTITY, GROUP_ENTITY } from '../utils/entity-types';
+import Login from './auth/login';
 
 loadIcons();
 
-export default () => (
-  <div>
-    <Navbar />
-  </div>
-);
+class App extends Component {
+  componentDidUpdate() {
+    const { fetched } = this.props;
+    if (this.props.authenticated && !fetched) {
+      this.props.fetchAllFieldConfigs();
+      this.props.fetchEntity(PERMISSION_ENTITY);
```

Comment:
entities will not depend on the field config fetch, every entity type should have it's own fetched property